include/glow/Partitioner/PartitionerBase.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_PARTITIONER_PARTITIONERBASE_H
+#define GLOW_PARTITIONER_PARTITIONERBASE_H
+
+#include "glow/Partitioner/PartitionerTypes.h"
+#include "glow/Support/Error.h"
+
+namespace glow {
+
+using namespace runtime;
+/// Given a module, partitions each of the its functions into multiple ones
+/// based on memory constraints and minimizes the communication cost.
+class PartitionerBase {
```

Comment:
let's clean up this interface a bit as discussed offline.

Reply:
Done.